src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -687,7 +677,14 @@ load_modules(sentry_value_t modules)
             continue;
         }
 
-        if (is_valid_elf_header((void *)(size_t)module.start)) {
+        // Module is appended if next module has different file name
+        if ((!last_module.file.len
+                && (!module.file.len || last_module.file.len != module.file.len
+                    || memcmp(last_module.file.ptr, module.file.ptr,
+                        module.file.len)))
+            || (!module.file.len || last_module.file.len != module.file.len
+                || memcmp(
+                    last_module.file.ptr, module.file.ptr, module.file.len))) {
```

Comment:
you should use `!sentry__slice_eq(module.file, last_module.file)` here to make this more readable.

Reply:
The key here is that the ELF check must be removed. Not all memory areas are valid ELF binaries. It doesn't make sense to search for ELF header of the next memory area as a condition to add a module to the list. Quite often there's anonymous memory areas between loaded libraries, and in such case the original code doesn't work.

Reply:
Hi @sc-aki. Can you provide an example of such a memory map (or just a few entries that visualize the problem)?

> Not all memory areas are valid ELF binaries.

But do they still contain valid ELF code referenced from the stack trace?

> Quite often there's anonymous memory areas between loaded libraries, and in such case the original code doesn't work.

Yes, in these cases, we would ignore that map entry. Are you saying that you have anonymous entries that contain code that your stack trace is referencing? How do you ensure the code is readable in the backend symbolication? 

Wouldn't this ignore anonymous map entries as well: https://github.com/getsentry/sentry-native/blob/master/src/modulefinder/sentry_modulefinder_linux.c#L677-L681

Reply:
> How do you ensure the code is readable in the backend symbolication?

Or is this change only meant for Android (where symbolication happens in the client)?